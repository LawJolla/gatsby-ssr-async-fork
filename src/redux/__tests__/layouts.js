const reducer = require(`../reducers/layouts`)
const { actions } = require(`../actions`)

describe(`Add layouts`, () => {
  it(`allows you to add layouts`, () => {
    const action = actions.createLayout(
      {
        component: `/whatever/index.js`,
      },
      { name: `test` }
    )
    const state = reducer(undefined, action)
    expect(action).toMatchSnapshot()
    expect(state).toMatchSnapshot()
  })

  it(`allows you to add layouts and specify their id`, () => {
    const action = actions.createLayout(
      {
        component: `/whatever/index.js`,
        id: `test-id`,
      },
      { name: `test` }
    )
    const state = reducer(undefined, action)
    expect(action).toMatchSnapshot()
    expect(state).toMatchSnapshot()
  })

  it(`allows you to add layout with context`, () => {
    const action = actions.createLayout(
      {
        component: `/whatever/index.js`,
        context: {
          title: `layout title`,
        },
      },
      { name: `test` }
    )
    const state = reducer(undefined, action)
    expect(action).toMatchSnapshot()
    expect(state).toMatchSnapshot()
  })

  it(`allows you to add multiple layouts`, () => {
    const action = actions.createLayout(
      {
        component: `/whatever/index.js`,
      },
      { name: `test` }
    )
    const action2 = actions.createLayout(
      {
        component: `/whatever/pizza.js`,
      },
      { name: `test` }
    )
    let state = reducer(undefined, action)
    state = reducer(state, action2)
    expect(state).toMatchSnapshot()
    expect(state.length).toEqual(2)
  })

  it(`allows you to delete ids`, () => {
    const action = actions.createLayout(
      {
        component: `/whatever/index.js`,
      },
      { name: `test` }
    )
    const action2 = actions.deleteLayout({ id: `index` })

    let state = reducer(undefined, action)
    state = reducer(state, action2)
    expect(state).toMatchSnapshot()
    expect(state.length).toEqual(0)
  })
})
