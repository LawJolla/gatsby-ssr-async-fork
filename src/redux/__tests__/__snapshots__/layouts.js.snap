// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add layouts allows you to add layout with context 1`] = `
Object {
  "payload": Object {
    "component": "/whatever/index.js",
    "componentChunkName": "component---whatever-index-js",
    "componentWrapperPath": "/.cache/layouts/index.js",
    "context": Object {
      "title": "layout title",
    },
    "id": "index",
    "internalComponentName": "Component-layout-Index",
    "isLayout": true,
    "jsonName": "layout-index.json",
    "pluginCreator___NODE": "Plugin test",
  },
  "plugin": Object {
    "name": "test",
  },
  "traceId": undefined,
  "type": "CREATE_LAYOUT",
}
`;

exports[`Add layouts allows you to add layout with context 2`] = `
Array [
  Object {
    "component": "/whatever/index.js",
    "componentChunkName": "component---whatever-index-js",
    "componentWrapperPath": "/.cache/layouts/index.js",
    "context": Object {
      "title": "layout title",
    },
    "id": "index",
    "internalComponentName": "Component-layout-Index",
    "isLayout": true,
    "jsonName": "layout-index.json",
    "pluginCreator___NODE": "Plugin test",
  },
]
`;

exports[`Add layouts allows you to add layouts 1`] = `
Object {
  "payload": Object {
    "component": "/whatever/index.js",
    "componentChunkName": "component---whatever-index-js",
    "componentWrapperPath": "/.cache/layouts/index.js",
    "context": Object {},
    "id": "index",
    "internalComponentName": "Component-layout-Index",
    "isLayout": true,
    "jsonName": "layout-index.json",
    "pluginCreator___NODE": "Plugin test",
  },
  "plugin": Object {
    "name": "test",
  },
  "traceId": undefined,
  "type": "CREATE_LAYOUT",
}
`;

exports[`Add layouts allows you to add layouts 2`] = `
Array [
  Object {
    "component": "/whatever/index.js",
    "componentChunkName": "component---whatever-index-js",
    "componentWrapperPath": "/.cache/layouts/index.js",
    "context": Object {},
    "id": "index",
    "internalComponentName": "Component-layout-Index",
    "isLayout": true,
    "jsonName": "layout-index.json",
    "pluginCreator___NODE": "Plugin test",
  },
]
`;

exports[`Add layouts allows you to add layouts and specify their id 1`] = `
Object {
  "payload": Object {
    "component": "/whatever/index.js",
    "componentChunkName": "component---whatever-index-js",
    "componentWrapperPath": "/.cache/layouts/test-id.js",
    "context": Object {},
    "id": "test-id",
    "internalComponentName": "Component-layout-TestId",
    "isLayout": true,
    "jsonName": "layout-test-id.json",
    "pluginCreator___NODE": "Plugin test",
  },
  "plugin": Object {
    "name": "test",
  },
  "traceId": undefined,
  "type": "CREATE_LAYOUT",
}
`;

exports[`Add layouts allows you to add layouts and specify their id 2`] = `
Array [
  Object {
    "component": "/whatever/index.js",
    "componentChunkName": "component---whatever-index-js",
    "componentWrapperPath": "/.cache/layouts/test-id.js",
    "context": Object {},
    "id": "test-id",
    "internalComponentName": "Component-layout-TestId",
    "isLayout": true,
    "jsonName": "layout-test-id.json",
    "pluginCreator___NODE": "Plugin test",
  },
]
`;

exports[`Add layouts allows you to add multiple layouts 1`] = `
Array [
  Object {
    "component": "/whatever/index.js",
    "componentChunkName": "component---whatever-index-js",
    "componentWrapperPath": "/.cache/layouts/index.js",
    "context": Object {},
    "id": "index",
    "internalComponentName": "Component-layout-Index",
    "isLayout": true,
    "jsonName": "layout-index.json",
    "pluginCreator___NODE": "Plugin test",
  },
  Object {
    "component": "/whatever/pizza.js",
    "componentChunkName": "component---whatever-pizza-js",
    "componentWrapperPath": "/.cache/layouts/pizza.js",
    "context": Object {},
    "id": "pizza",
    "internalComponentName": "Component-layout-Pizza",
    "isLayout": true,
    "jsonName": "layout-pizza.json",
    "pluginCreator___NODE": "Plugin test",
  },
]
`;

exports[`Add layouts allows you to delete ids 1`] = `Array []`;
